server/src/main/java/de/hso/badenair/service/account/AccountService.java

```diff
@@ -1,43 +1,73 @@
 
 @Service
 @RequiredArgsConstructor
 public class AccountService {
 
-    private final AccountDataRepository accountDataRepository;
-
-    public AccountDataDto getAccountData(String userId) {
-        return accountDataRepository.findByCustomerUserId(userId)
-            .map(AccountDataMapper::mapToDto)
-            .orElse(AccountDataDto.builder().build());
-    }
-
-    @Transactional
-    public void updateAccountData(String userId, UpdateAccountDataDto dto) {
-        final AccountData accountData = accountDataRepository.findByCustomerUserId(userId)
-            .orElseGet(this::getEmptyAccountData);
-
-        accountData.setCustomerUserId(userId);
-        accountData.setBirthday(dto.getBirthDate());
-        accountData.setStreet(dto.getStreet());
-        accountData.setZipCode(dto.getZipCode());
-        accountData.setCity(dto.getPlaceOfResidence());
-        accountData.setCardOwner(dto.getCardOwner());
-        accountData.setCardNumber(dto.getCardNumber());
-        accountData.setCvv(dto.getCheck());
-        accountData.setExpirationDate(dto.getInvalidationDate());
-    }
-
-    private AccountData getEmptyAccountData() {
-        return accountDataRepository.save(AccountData.builder().build());
-    }
+	private final AccountDataRepository accountDataRepository;
+
+	private final BookingRepository bookingRepository;
+
+	public AccountDataDto getAccountData(String userId) {
+		return accountDataRepository.findByCustomerUserId(userId)
+				.map(AccountDataMapper::mapToDto)
+				.orElse(AccountDataDto.builder().build());
+	}
+
+	@Transactional
+	public void updateAccountData(String userId, UpdateAccountDataDto dto) {
+		final AccountData accountData = accountDataRepository
+				.findByCustomerUserId(userId)
+				.orElseGet(this::getEmptyAccountData);
+
+		accountData.setCustomerUserId(userId);
+		accountData.setBirthday(dto.getBirthDate());
+		accountData.setStreet(dto.getStreet());
+		accountData.setZipCode(dto.getZipCode());
+		accountData.setCity(dto.getPlaceOfResidence());
+		accountData.setCardOwner(dto.getCardOwner());
+		accountData.setCardNumber(dto.getCardNumber());
+		accountData.setCvv(dto.getCheck());
+		accountData.setExpirationDate(dto.getInvalidationDate());
+	}
+
+	private AccountData getEmptyAccountData() {
+		return accountDataRepository.save(AccountData.builder().build());
+	}
+
+	public List<Booking> getBookings(String customerUserId) {
+		List<Booking> bookings = bookingRepository
+				.findAllByCustomerUserId(customerUserId);
+		List<Booking> bookingsFiltered = new ArrayList<Booking>();
+
+		// Filter out bookings of the past
+		for (Booking booking : bookings) {
+			if (booking.getFlight().getStartDate()
+					.plusHours(booking.getFlight().getScheduledFlight()
+							.getStartTime().getHour())
+					.plusMinutes(booking.getFlight().getScheduledFlight()
+							.getStartTime().getMinute())
+					.isAfter(OffsetDateTime.now())) {
+				bookingsFiltered.add(booking);
+			}
+		}
+
+		return bookings;
```

Comment:
```suggestion
		return bookingsFiltered;
```
Sollte hier nicht `bookingsFiltered` zurückgegeben werden ?

Reply:
Ja, mein Fehler.